### ISSUE TITLE
Enums are not supported in d.ts files

### ISSUE THREAD

**Issue by dzmitry-penda** [angular/angular-cli#8068]
### Bug Report or Feature Request (mark with an `x`)
```
- [x ] bug report -> please search issues before submitting
- [ ] feature request
```

### Versions.
@angular/cli: 1.4.7
node: 6.11.0
os: win32 x64
@angular/animations: 4.4.5
@angular/common: 4.4.5
@angular/compiler: 4.4.5
@angular/core: 4.4.5
@angular/forms: 4.4.5
@angular/http: 4.4.5
@angular/platform-browser: 4.4.5
@angular/platform-browser-dynamic: 4.4.5
@angular/router: 4.4.5
@angular/cli: 1.4.7
@angular/compiler-cli: 4.4.5
@angular/language-service: 4.4.5
typescript: 2.3.4

### Repro steps.

1.  Run `ng new` to create a new app
2.  Create file `enum.d.ts` with some enum like `export enum Sample {  Yes,  No }`
3.  Add reference to `enum.d.ts` in app.component.ts and use it somehow like `sample = Sample.Yes;`
4. Try to run the app 

### The log given by the failure.
ERROR in ./src/app/enum.d.ts
Module build failed: Error: Debug Failure. False expression: Output generation failed
    at Object.assert (F:\GIT\test-enum\node_modules\typescript\lib\typescript.js:3329:23)
    at Object.transpileModule (F:\GIT\test-enum\node_modules\typescript\lib\typescript.js:80449:18)
    at TypeScriptFileRefactor.transpile (F:\GIT\test-enum\node_modules\@ngtools\webpack\src\refactor.js:166:27)
    at Promise.resolve.then.then.then.then (F:\GIT\test-enum\node_modules\@ngtools\webpack\src\loader.js:523:37)
    at process._tickCallback (internal/process/next_tick.js:109:7)
 @ ./src/app/app.component.ts 8:0-34
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:3333 ./src/main.ts

### Desired functionality.
No compilation error

**Comment by clydin** [comment]
`.d.ts` files only provide type information and are not compiled.  The enum needs to be defined within an actual source file.

**Comment by dzmitry-penda** [comment]
Yes, the example is not correct. But `const enum` will be inlined instead of references to its members and will not produce any output. So we should be able to declare const enums in `.d.ts` files. For instance, it can be done in `ts-loader`.

**Comment by hansl** [comment]
Hi @dzmitry-penda,

If I understand correctly, this issue is already fixed in master. However there is catch; because it needed an entire new pipeline to be done, we only did the work to fix it in CLI 1.5, when using Angular 5. This is because supporting this in either 1.4 or Angular 4 would have required a lot more conditions (e.g. TypeScript 2.4 is needed).

You can either upgrade to using the RC of CLI 1.5 and the RC of Angular 5, or wait a bit longer to use the final versions of both.

Hope this helps!
